model/config.go

```diff
@@ -1553,6 +1559,10 @@ func (o *Config) IsValid() *AppError {
 		}
 	}
 
+	if *o.SamlSettings.Enable || *o.LdapSettings.Enable {
+		*o.TeamSettings.TeammateNameDisplay = SHOW_FULLNAME
+	}
```

Comment:
This is in the wrong spot I think, it should be done in SetDefaults() inside the `if o.TeamSettings.TeammateNameDisplay == nil {` statement